azurerm/internal/services/network/application_gateway_resource.go

```diff
@@ -2332,6 +2342,10 @@ func flattenApplicationGatewayHTTPListeners(input *[]network.ApplicationGatewayH
 				output["require_sni"] = *sni
 			}
 
+			if fwp := props.FirewallPolicy; fwp != nil {
```

Comment:
there's a crash here if `id` is nil:

```suggestion
			if fwp := props.FirewallPolicy; fwp != nil && fwp.ID != nil {
```